hw2/05-form.js

```diff
@@ -1 +1,123 @@
+// to take place.
+document.getElementById("dataForm").addEventListener("submit", handleSubmit);
+
+// I get the modalButtonClose element, and use this to trigger closing the modal
+// when it's open.
+document
+    .getElementById("modalButtonClose")
+    .addEventListener("click", hideModal);
+
+// Easier to just keep track of this as a global const since it is updated in
+// Multiple places.
+const myModal = new bootstrap.Modal(document.getElementById("myModal"));
+
+const nameToModal = function fullNameToTheModal(fullName) {
+    const modalBody = document.getElementById("modalBody");
+    const fullNamePara = document.createElement("p");
+    fullNamePara.innerText = `Full name: ${fullName}`;
+    modalBody.appendChild(fullNamePara);
+    return;
+};
+
+const emailToModal = function emailToTheModal(email) {
+    const modalBody = document.getElementById("modalBody");
+    const emailPara = document.createElement("p");
+    emailPara.innerText = `Email: ${email}`;
+    modalBody.appendChild(emailPara);
+    return;
+};
+
+const regStatToModal = function registeredStatusToModal(regStatus) {
+    const regStatusPara = document.createElement("p");
+    regStatusPara.innerText = `Registration Status: ${regStatus}`;
+    modalBody.appendChild(regStatusPara);
+    return;
+};
+
+const classToModal = function classTakenToModal(className) {
+    if (className.length > 0) {
+        const courseTakenPara = document.createElement("p");
+        courseTakenPara.innerText = `Course completed - ${className}`;
+        modalBody.appendChild(courseTakenPara);
+    }
+};
+
+const moreToModal = function additionalToModal(addtionals) {
+    const additionalCommentsPara = document.createElement("p");
+    additionalCommentsPara.innerText = `Additional Comments: ${addtionals}`;
+    modalBody.appendChild(additionalCommentsPara);
+    return;
+};
+
+function displayModal(allData) {
+    // Get the main body of the modal really quick.
+    const modalBody = document.getElementById("modalBody");
+
+    // Reset modalBody in the event it still has data
+    modalBody.innerHTML = "";
+
+    // Call a bunch of functions to do our work to add to the modal.
+    // I'm doing this a bunch of times because there's additional text I'd like
+    // to dd
+    nameToModal(allData[0]);
+    emailToModal(allData[1]);
+    regStatToModal(allData[2]);
+    classToModal(allData[3]);
+    classToModal(allData[4]);
+    classToModal(allData[5]);
+    moreToModal(allData[6]);
+
+    // Okay, we've built it with all the content, lets show it off!
+    myModal.show();
+}
+
+// Way easier to hide it than showing it...
+function hideModal() {
+    document.getElementById("modalBody").innerHTML = "";
+    myModal.hide();
+}
+
+// Write a few ternaries so they are easier to read on their own.
+// Each either returns the value in the input (the class name) or it returns
+// an empty string. We check/use this later when building the modal.
+const getPL = function getProgramingLanguage() {
```

Comment:
This highlights the power of using the ternary operator. As you mentioned, the Airbnb documentation states in rule **15.6** to not be nested and be single-line expressions. I know that prettier automatically formats them this way and I find it easier to read them like this over single-line expressions.

Reply:
Agreed! This certainly is easier to read, but you can't always get away with a single line too. I suspect what someone would say who wrote that style guide would be like "if you can't fit it into a single line, just turn it into an if/else" but then you're just sitting there with even more lines of code that do the exact same thing. Prettier taking over and formatting it nicely is really great here.